hermeto/core/package_managers/cargo/main.py

```diff
@@ -220,26 +253,72 @@ def _run_cmd_watching_out_for_lock_mismatch(
             raise
 
 
-def _generate_sbom_components(package_dir: RootedPath) -> chain[Component]:
+def _find_local_packages(package_dir: RootedPath) -> dict[str, str]:
+    """Find local packages in the Cargo.toml file and return their subpaths."""
+    parsed_toml = _parse_toml_project_file(package_dir.path / "Cargo.toml")
+
+    result = {}
+
+    runtime_deps = parsed_toml.get("dependencies", {})
+    # https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html
+    patched_deps = parsed_toml.get("patch", {}).get("crates-io", {})
+    all_deps = {**runtime_deps, **patched_deps}
+
+    for name, dep_info in all_deps.items():
+        if isinstance(dep_info, dict) and "path" in dep_info:
+            result[name] = dep_info["path"]
+
+    return result
+
+
+def _generate_sbom_components(package_dir: RootedPath) -> list[Component]:
     """Generate SBOM components from Cargo.lock and for the main package."""
     parsed_lockfile = _parse_toml_project_file(package_dir.path / "Cargo.lock")
-    packages = parsed_lockfile.get("package", [])
 
+    all_packages = parsed_lockfile.get("package", [])
+    local_packages = _find_local_packages(package_dir)
     main_package_name, main_package_version = _resolve_main_package(package_dir)
-    is_a_dep = lambda p: p["name"] != main_package_name
+
     try:
         vcs_url = get_repo_id(package_dir.root).as_vcs_url_qualifier()
     except NotAGitRepo:
         # Could become invalid when directories are swapped for nested package managers
         vcs_url = None
-    deps_components = (
-        CargoPackage(**p, vcs_url=vcs_url).to_component() for p in packages if is_a_dep(p)
-    )
-    main_component = CargoPackage(
-        name=main_package_name, version=main_package_version, vcs_url=vcs_url
-    ).to_component()
 
-    components = chain((main_component,), deps_components)
+    components = []
+
+    for pkg in all_packages:
+        pkg_name = pkg.get("name")
+        pkg_version = pkg.get("version")
+
+        if pkg_name == main_package_name:
+            components.append(
+                LocalCargoPackage(
+                    name=main_package_name,
+                    version=main_package_version,
+                    vcs_url=vcs_url,
+                    subpath=str(package_dir.path.relative_to(package_dir.root)),
+                ).to_component()
+            )
+        elif pkg_name in local_packages.keys():
```

Comment:
Non-blocking observation: having a `is_local_package()` defined somewhere would have slightly improved the flow. Returning it from `_find_local_packaged()` (properly renamed) could be done. Bending the rules a little and agreeing on the helper to return either an empty string or a path could result in something like this:
```python
elif subpath := is_local_package(pcg_name):
   ...
   subpath=subpath,
   ...
```